testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -690,9 +691,16 @@ public void testSelectInTransaction()
     @Test
     public void testSelectVersionOfNonExistentTable()
     {
+        String tableName = "foo_" + randomNameSuffix();
+
+        if (!hasBehavior(SUPPORTS_VERSIONED_TABLE)) {
```

Comment:
We could change the assertion to `hasMessageMatching` instead. 

Introducing `SUPPORTS_VERSIONED_TABLE` requires more work. For instance, javadoc in testTrySelectTableVersion needs update.